include/bluetooth/mesh/access.h

```diff
@@ -732,6 +732,85 @@ struct bt_mesh_comp {
 	struct bt_mesh_elem *elem; /**< List of elements. */
 };
 
+/**
```

Comment:
Does it worth to move these new functions to a separate file, e.g. model_utils.h?

Reply:
I'm not sure why you shied away from calling this "client" like I originally suggested, since I don't think this use case exists for server models (they always respond to requests but never wait on anything from a client). So I'd have been fine with a cli.c and calling the API like I originally suggested. @trond-snekvik any thoughts on the naming and placement?

Reply:
I'm kind of tempted to suggest moving this to a models.h file, along with the model IDs, since neither concept is really part of access. I gotta do a proper review tomorrow though

Reply:
I also think this should be in a separate file although I'm not sure if it should be models.h as this will be used by client models only. I agree that `client` should be in the naming scheme. Anyway, this change is very welcome.

Reply:
The naming is based on the original downstream API in Nordic's SDK: https://github.com/nrfconnect/sdk-nrf/blob/master/subsys/bluetooth/mesh/model_utils.h#L70-L122. That doesn't dictate that this API needs to follow the same convention, but I would prefer not to tie it to the client, as this pattern could show up in future non-client models. I don't feel very strongly about this though.

Reply:
@trond-snekvik why was this done to the downstream tree to begin with, rather than making this refactoring in the upstream tree? Seems like that would have benefitted both parties :)

Reply:
Anyway, I don't have a strong opinion on the naming, but passing the msg_context is IMO a bit debatable - it also seems contrary to the general inclination of trying to make this API applicable to a wider set of use cases.

Reply:
> it also seems contrary to the general inclination of trying to make this API applicable to a wider set of use cases

Just to clarify what I mean by this: what if you have some use case where you don't have a msg_ctx. E.g. you get the request (through some OOB means) to cancel the operation for a given address. Anyway, I guess all this is fairly theoretical, as is the desire to handle cases where you don't wait for any specific parameter values but just a response, or do you have a case for that in the downstream tree?.

Reply:
> why was this done to the downstream tree to begin with

Not sure to be honest. I think I was putting it in downstream to get the ball rolling so I could upstream it later. Then "later" never happened (until now) :slightly_smiling_face: 

After some more consideration, I want to push for adding models.h, and put this and the model IDs in it. The model IDs and this client-utility introduces concepts that are distinctively above the access layer. Perhaps the `bt_mesh_model_msg` macros and function does as well? Although a large portion of the functions in access.h are named `bt_mesh_model_...`, they're all implementing access-specific functionality, and makes no assumption about the sort of models you would implement. 

You've won me over with the ctx argument, comparing directly on the `ctx->addr` makes sense.

I believe the node reset function in config client is a case where having no user data makes sense, as the node_reset_status message doesn't contain any parameters. It's still interesting to wait for this response, as we may want to wait for the node to be reset fully before e.g. reprovisioning it. The current implementation takes a pointer to a bool, but it's never set to false, so it's kind of useless.